modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/scroll/ElasticSearchScroll.java

```diff
@@ -48,13 +51,17 @@ public boolean hasNext() {
     @Override
     protected boolean fetch() {
         ElasticSearchService ess = Framework.getService(ElasticSearchService.class);
-        if (esScroll == null) {
-            esScroll = ess.scroll(new NxQueryBuilder(session).nxql(request.getQuery())
-                                                             .limit(request.getSize())
-                                                             .onlyElasticsearchResponse(),
-                    request.getTimeout().toSeconds());
-        } else {
-            esScroll = ess.scroll(esScroll);
+        try {
+            if (esScroll == null) {
+                esScroll = ess.scroll(new NxQueryBuilder(session).nxql(request.getQuery())
+                                .limit(request.getSize())
+                                .onlyElasticsearchResponse(),
+                        request.getTimeout().toSeconds());
+            } else {
+                esScroll = ess.scroll(esScroll);
+            }
+        } catch (ElasticsearchException e) {
+            throw new NuxeoException("Elastic scroll failure", e);
```

Comment:
Wouldn't it be interesting to add the NXQL query that failed in the exception message?

Reply:
yes we could print the scroll query that contains the nxql